mod_dynamic_spawns/config.nut

```diff
@@ -84,19 +84,20 @@
 	if (_def instanceof _dataset.BaseClass)
 		return (clone _def).init();
 
-	if ("ID" in _def)
+	if ("BaseID" in _def)
 	{
-		local baseDef = _dataset.LookupMap[_def.ID];
+		local baseDef = _dataset.LookupMap[_def.BaseID];
 		::DynamicSpawns.__setClass(baseDef.Class, _def);
 		local ret = _def.Class(baseDef);
 		ret.copyDataFromDef(_def);
+		if (ret.ID == baseDef.ID) ret.ID += ret + "";
```

Comment:
What is happening here?
Are you overwriting the ID with some sort of hash value of a table?

Reply:
It's the memory address of the table. It looks like `(table 0x34ADF324)`. This ensures that the ID is unique, so that anything accessing it by ID during a `spawnProcess` always has each thing with a unique ID.